documentation/docs/specs/revocation_bitmap_2022.md

```diff
@@ -0,0 +1,182 @@
+
+This specification describes an on-tangle mechanism for publishing the revocation status of [Verifiable Credentials](../concepts/verifiable_credentials/overview) embedded in an issuer's DID document.
+
+## Introduction
+
+Revocation gives an issuer the possibility to invalidate a credential they issued before its natural expiration date. To that end, issuers embed an identifier in the credential's `credentialStatus` field and verifiers can lookup that identifier in a list to check whether the credential is valid. This document specifies a mechanism of embedding such a list, in form of a bitmap, in an issuer's DID document, where each bitmap index corresponds to a credential the issuer has issued. This mechanism is space-efficient, enables a verifier to check a credential's status in a privacy-preserving manner and without requiring additional lookups or external resources.
+
+## Revocation Bitmap Concept
+
+The revocation status of a verifiable credential is expressed as a binary value. The issuer keeps a bitmap of indices corresponding to verifiable credentials it has issued. If the binary value of the index in the bitmap is 1 (one), the verifiable credential is revoked, if it is 0 (zero) it is not revoked.
+
+## Data Model
+
+### Revocation Bitmap Status
+
+For an issuer to enable verifiers to check the status of a verifiable credential, the [`credentialStatus`](https://www.w3.org/TR/vc-data-model/#status) property MUST be specified with the following properties:
+
+- **id**: A [DID URL](https://www.w3.org/TR/did-core/#did-url-syntax) that can be resolved to a service of the issuer which contains the revocation bitmap;
+- **type**: the string `"RevocationBitmap2022"`;
+- **revocationBitmapIndex**: an unsigned, 32-bit integer expressed as a string, which is the index of the verifiable credential's revocation status in the issuer's revocation bitmap;
+
+### Revocation Bitmap Service
+
+In order to allow verfiers to check the status of a verifiable credential, the DID document of the issuer may contain a service that MUST have the following properties:
+
+- **id**: a URL identifying the revocation bitmap;
+- **type**: the string `"RevocationBitmap2022"`;
+- **serviceEndpoint**: a `data` URL [[RFC2397](https://datatracker.ietf.org/doc/html/rfc2397)] embedding a revocation bitmap generated according to the [bitmap generation algorithm](#Bitmap-Generation-Algorithm). The `<mediatype>` MUST be `application/octet-stream` and the `base64` attribute MUST be set.
+
+## Algorithms
+
+The following algorithms describe how to generate and validate revocation bitmaps.
+
+### Bitmap Generation Algorithm
+
+The following process MUST be followed when producing a `RevocationBitmap2022`:
+
+1. Let **bitmap** be a [_roaring bitmap_](https://roaringbitmap.org/) where each bit is initialized to 0 (zero).
+2. For each revoked credential with an **identifier** not exceeding an unsigned, 32-bit integer, set the bit in **bitmap** at index **identifier** to 1.
```

Comment:
I would be careful using `identifier` here, the terminology can be confused with `id`. Prefer `index` perhaps?

```suggestion
1. Let **bitmap** be a [_roaring bitmap_](https://roaringbitmap.org/) where each bit is initialized to 0.
2. For each revoked credential with an **index** not exceeding an unsigned, 32-bit integer, set the corresponding bit in **bitmap** at **index** to 1.
```